packages/expo-image/ios/ImageView.swift

```diff
@@ -210,14 +210,28 @@ public final class ImageView: ExpoView {
       log.debug("Loading the image has been canceled")
       return
     }
+
+    // First we need to see if the image is animated. SDAnimatedImage might not properly render some image formats.
+    // If the image is animated, we want to use the data and create an SDAnimatedImage with that data.
+    // SDAnimatedImage extends UIImage so we should be able to do everything that we were doing before still.
+    var animated = false
+    var image = image
+
+    if image?.sd_isAnimated != nil, let data = data {
+      animated = true
+      image = SDAnimatedImage(data: data)
+    }
```

Comment:
In general, maybe we can make a separate function for that, so we don't add too much logic in this `imageLoadCompleted` function? 🤔

Reply:
Added a function inside ImageUtils to do this.